### DISCUSSION TITLE
Queue Job Response.

### DISCUSSION THREAD

**Discussion by R00tes** [filamentphp/filament#1859] [Help]
Hello,
Again I love your work guys! I do own a Dance school (Not much Website experience) and your code makes all so easy. Thanks to make it easy for amateurs like me. 

OK, to the question:
I have this Action in my Table that calls a JOB and I would like to know if is possible to have the Icon change to Loading until the Job is finished and maybe refresh the Table after.

```

protected function getTableActions(): array
    {
        return [
            \Filament\Tables\Actions\IconButtonAction::make(__('Paid'))
                ->icon('heroicon-o-cash')
                ->action(function (Registration $record): void {
                    PaidRegistration::dispatch($record);      <------- QUEUE JOB
                    $this->notify('primary', 'Student Paid Registration', isAfterRedirect: false);
                })->color('primary'),
        ];
    }

```

I did found the Queue:after but not sure how to call Filament after.

```
    /**
     * Bootstrap any application services.
     *
     * @return void
     */
    public function boot()
    {
        Queue::after(function (JobProcessed $event) {
          //
        });
    }
```

PS: As soon as the corona pass and my dance school gets a bit better I would love to give a donation, just I didn't find the link if you can add it will be great.

Thanks again for all the support.
Inaki

**Comment by ryangjchandler** [comment]
@R00tes I assume your `PaidRegistration` job is being run on a queue (in the background)? If that's the case, it's not going to be super straight forward.

**Reply by R00tes** [reply]
Yes, Is in the Job/queue and I use Redis Driver,  (The queue is working perfectly) .. i just want to know how to call the table and IconButtonAction after the job is done.

**Reply by danharrin** [reply]
I think you'd have to use websockets for that, since queued jobs run asynchronously to your actual app. Your easiest route would be to not queue this job at all.

**Reply by R00tes** [reply]
Now I know how it works.. I been looking into Websockets... Thanks for pointing me the direction.
